src/inmemory_cache.rs

```diff
@@ -6,19 +6,22 @@ use crate::{
     block::Block, db_state::SsTableId, filter::BloomFilter, flatbuffer_types::SsTableIndexOwned,
 };
 
+/// The cached block types.
 #[derive(Clone)]
-pub(crate) enum CachedBlock {
+pub enum CachedBlock {
     Block(Arc<Block>),
     Index(Arc<SsTableIndexOwned>),
     Filter(Arc<BloomFilter>),
 }
 
+/// The type of the in-memory cache.
 #[derive(Clone, Copy)]
 pub enum CacheType {
     Moka,
     Foyer,
 }
 
+/// The options for the in-memory cache.
 #[derive(Clone, Copy)]
 pub struct InMemoryCacheOptions {
```

Comment:
Same here. We can have per-implementation options now (e.g. MokaCacheOptions, FoyerCacheOptions)

Reply:
I've implemented two separated options structs, although they have the same fields at the moment.